lib/string.js

```diff
@@ -430,4 +437,31 @@ internals.String.prototype.trim = function () {
     return obj;
 };
 
+
+internals.String.prototype.replace = function (pattern, replacement) {
+
+    Hoek.assert(pattern instanceof RegExp, 'pattern must be a RegExp');
```

Comment:
It would be nice to also be able to provide a string and transform it to a regexp with flag g to replace everything.